flume-ng-channels/flume-kafka-channel/src/main/java/org/apache/flume/channel/kafka/KafkaChannel.java

```diff
@@ -564,15 +560,22 @@ protected void doCommit() throws InterruptedException {
                   ex);
         }
       } else {
+        // event taken ensures that we have collected events in this transaction
+        // before committing
         if (consumerAndRecords.get().failedEvents.isEmpty() && eventTaken) {
+          logger.trace("About to commit batch.");
           long startTime = System.nanoTime();
           consumerAndRecords.get().commitOffsets();
           long endTime = System.nanoTime();
           counter.addToKafkaCommitTimer((endTime - startTime) / (1000 * 1000));
-          consumerAndRecords.get().printCurrentAssignment();
+          logger.debug("{}", consumerAndRecords.get().getCommittedOffsetsString());
+        }
+
+        int takes = events.get().size();
```

Comment:
Maybe? I don't have a strong opinion on it and would prefer to make as few changes as possible.